hal/armv7a/cpu.c

```diff
@@ -19,6 +19,7 @@
 #include "../cpu.h"
 #include "../string.h"
 #include "../spinlock.h"
+#include "../../proc/process.h"
```

Comment:
top -> down module referencing (i.e. using proc API in HAL) is forbidden. `tls_t` definition should be located in HAL (and named `hal_tls_t`), this will solve this issue.